libs/libcommon/src/libcommon/viewer_utils/index_utils.py

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+from os import makedirs
+from pathlib import Path
+
+from libcommon.storage import StrPath
+
+DATASET_SEPARATOR = "--"
+INDEX_DIR_MODE = 0o755
+
+
+def create_index_dir_split(dataset: str, config: str, split: str, index_directory: StrPath) -> Path:
```

Comment:
> is it temporary? do we delete it later?

Yes, I call     remove_dir(dir_path) in line 226, but will better apply the same approach as in `datasets library cache` (create a random directory to avoid concurrency in pre-compute and delete it in post-compute)

Reply:
OK, let's try to use the same code for both (even if the datasets library cache has some additional code to set and unset the config variable